### ISSUE TITLE
STIX Patterning language needs a function to measure the length of a property

### ISSUE THREAD

**Issue by treyka** [oasis-tcs/cti-stix2#51]
In other words, is an array or a dict present but empty? Currently no way to determine this.

**Comment by gtback** [comment]
I can see the value of this function, but am reluctant to include it in STIX 2.1 (as with pretty much any other change to patterning 😉 ). I think it would be better to have a more comprehensive solution to handling functions on properties, than making a one-off solution for length.

**Comment by ikiril01** [comment]
I can see the value of this as well (e.g., testing for the number of values under a registry key), but wouldn't actually specifying an empty array/dict as part of an object be invalid STIX?

**Comment by treyka** [comment]
@ikiril01 If you're evaluating a pattern against STIX Observed Data, then an empty array or dict would be invalid STIX. But consider that there's a huge swath of use cases for STIX Patterning where Observed Data doesn't enter the picture.

**Comment by ikiril01** [comment]
@treyka good point! I guess this also brings up the question - how do you represent in SCO the absence of a particularly property? Going back to my previous example, say a registry key with no values?

**Comment by jordan2175** [comment]
We talked about this on 2019-06-05 and we should try and get this done for 2.1. But it needs someone to do the work.
